// react-animation-composer/src/shadow_window.js
import React from 'react';
import ReactDOM from 'react-dom';

export default class ShadowWindow extends React.Component {
  static defaultProps = {
    transition: '',
    animation: ''
  };
  static propTypes = {
    transition: React.PropTypes.string.isRequired,
    animation: React.PropTypes.string.isRequired
  };
  isOpening(nextProps) {
    var props = nextProps || this.props;
    return props.transition === 'opening' ||
           (props.transition === 'opened' && props.animation === 'immediate');
  };
  isClosed(nextProps) {
    var props = nextProps || this.props;
    return props.transition === 'closed';
  };
  shouldComponentUpdate(nextProps, nextState) {
    // We are the gap river between XWindow and the content.
    // We will never update whenever XWindow got updated by the XWindowManager.
    if (this.isOpening(nextProps)) {
      return true;
    } else {
      return false;
    }
  };
  render() {
    return <div className="shadow-window">{this.props.children}</div>;
  }
};
